src/frontend/src/components/Posts/Posts.jsx

```diff
@@ -26,19 +28,48 @@ const useStyles = makeStyles(() => ({
   },
 }));
 
+const usePrevious = (value) => {
```

Comment:
It looks like you're trying to do https://usehooks.com/usePrevious/ here, but not correctly.  Your effect is going to run non-stop, you need to add a dependency array.  Also, this hook needs to be in `hooks/` and also, it probably exists as something we can import.

Reply:
Available in https://github.com/streamich/react-use/blob/master/docs/usePrevious.md